include/Dialect/TensorExt/IR/TensorExtOps.td

```diff
@@ -0,0 +1,40 @@
+#ifndef INCLUDE_DIALECT_TENSOREXT_IR_TENSOREXTOPS_TD_
+#define INCLUDE_DIALECT_TENSOREXT_IR_TENSOREXTOPS_TD_
+
+include "include/Dialect/TensorExt/IR/TensorExtDialect.td"
+include "mlir/IR/BuiltinAttributes.td"
+include "mlir/IR/CommonTypeConstraints.td"
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+
+class TensorExt_Op<string mnemonic, list<Trait> traits = []> :
+        Op<TensorExt_Dialect, mnemonic, traits> {
+  let cppNamespace = "::mlir::heir::tensor_ext";
+}
+
+def TensorExt_RotateOp : TensorExt_Op<"rotate", [Pure, AllTypesMatch<["tensor", "output"]>]> {
+  let summary = "Rotate a tensor some number of indices left.";
+  let description = [{
+    This op represents a left-rotation of a tensor by given number of indices.
```

Comment:
This may be with wrap-around right? I was going to comment that I wonder if a canonicalizer or validator should enforce / re-write wrap around indices with ones within the range, but I think that may not be possible for variable rotations.

Reply:
It is with wrap-around, correct. I don't think we need to worry about this though, unless we get to a lowering that actually implements rotate in terms of a loop.